### PR TITLE
Tinkerbell metadata url

### PR DESCRIPTION
**Pull request by panktishah26**
*Issue #, if available:*
Populate metadata_url based on tinkerbellIP.

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/tinkerbell/tinkerbell.go [hunk 1]
```diff
@@ -117,6 +117,8 @@ func NewProvider(
 		}
 	}
 
+	tinkerbellIp, _ := networkutils.GetLocalIP()
```
Comment (chrisdoherty4) [inline pkg/providers/tinkerbell/tinkerbell.go:120]: Any reason we aren't checking the error? I know this won't be the final code but we should still check the error.

Comment (panktishah26) [reply pkg/providers/tinkerbell/tinkerbell.go:120]: True, NewProvider construct does not return error so I had to ignore it. Should I put this somewhere else?

Comment (abhinavmpandey08) [reply pkg/providers/tinkerbell/tinkerbell.go:120]: you can get it in the provider factory and inject it from there to the provider

**Review by abhinavmpandey08 — APPROVED** [review]
/approve

**Pull request merged**